halo2_proofs/examples/shuffle.rs

```diff
@@ -0,0 +1,356 @@
+use ff::BatchInvert;
```

Comment:
Shall we add the example in another PR to have a more clear view of diffs? Unless this example targets the ZK-nonZK feature heavily.

Reply:
Make sense, this example just wants to make sure we can do wrap around trick to save some constraint and degree, but indeed it makes more sense to have it after challenge API is ready.